src/createServer.js

```diff
@@ -2,7 +1,43 @@
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
 
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer(async (req, res) => {
+    const { pathname } = new URL(req.url || '', `http://${req.headers.host}`);
+    const requestedPath = pathname.replace('/file', '') || 'index.html';
+    const realPath = path.join(__dirname, '..', 'public', requestedPath);
+
+    res.setHeader('Content-Type', 'text/plain');
+
+    if (!pathname.startsWith('/file')) {
```

Comment:
The condition `!pathname.startsWith('/file')` might not be sufficient if the intention is to handle routes that should start with '/file/'. Consider using `!pathname.startsWith('/file/')` to ensure the pathname starts with '/file/' exactly, preventing unintended matches.